### DISCUSSION TITLE
Query param of enum type accept enum name instead of enum value

### DISCUSSION THREAD

**Discussion by aliasgar55** [fastapi/fastapi#9302] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python


### Description

Create a Enum class in python
class TestEnum(Enum):
 name = "value"
 ```

Create a endpoint in and add Query() of type TestEnum
Visit the swagger docs, it show the availble options in param as name 
The above behavior contradicts with the behavior Contradicts if we add the same type as a body field
The endpoint will accept value instead of name in the Body

### Operating System

Linux

### Operating System Details

```Distributor ID: Pop
Description:    Pop!_OS 22.04 LTS
Release:        22.04
Codename:       jammy
 Static hostname: pop-os
       Icon name: computer-laptop
         Chassis: laptop
      Machine ID: b12731022c42a5ce6d5755de63fe20df
         Boot ID: 050d6f39271149dc9c209ee0e060b7ac
Operating System: Pop!_OS 22.04 LTS
          Kernel: Linux 6.1.11-76060111-generic
    Architecture: x86-64
```

### FastAPI Version

0.81.0

### Python Version

3.10.6

**Comment by mde-pach** [comment]
Can this can help you ?
https://fastapi.tiangolo.com/tutorial/path-params/?h=enum+value#get-the-enumeration-value

**Reply by aliasgar55** [reply]
Point is not about having a workaround
Instead providing a consistent interface
Enum in body and query serializes differently

**Reply by aliasgar55** [reply]
But how can I change what values the API accepts

**Comment by aliasgar55** [comment]
Is there a way to provide a encoder function in the Query  function

**Comment by bfontaine** [comment]
The behavior might have changed since because I’m using FastAPI 0.95.2 and the Swagger doc does show the value as a parameter, not the name.
